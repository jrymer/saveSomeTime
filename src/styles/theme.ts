// https://colorpalettes.net/color-palette-3983/
export const theme = {
  primary: '#4a919e',
  secondary: '#bed3c3',
  accent: '#ebaca2',
  warning: '#ce6a6b',
  dark: '#212e53',
  disabled: '#9096A9',
  font: '#FFF'
};
