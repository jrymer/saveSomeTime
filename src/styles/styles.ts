import { StyleSheet } from 'react-native';

import { theme } from './theme';

export const button = StyleSheet.create({
  roundButton: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 20,
    marginTop: 10,
    width: 320,
    height: 40
  }
});

export const text = StyleSheet.create({
  whiteText: {
    color: theme.font
  },
  warningText: {
    color: theme.warning
  }
});
