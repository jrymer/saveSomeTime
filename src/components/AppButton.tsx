import React from 'react';
import { Text, TextStyle, TouchableOpacity, ViewStyle } from 'react-native';

import { button, text } from '../styles';

interface IProps {
  label: string;
  buttonStyles?: ViewStyle;
  textStyles?: TextStyle;
  onPress?: () => void;
  disabled?: boolean;
}

const defaultProps: Partial<IProps> = {
  disabled: false
};

export class AppButton extends React.Component<IProps> {
  private static defaultProps = defaultProps;

  constructor(props: IProps) {
    super(props);
  }

  public render() {
    const {
      buttonStyles,
      textStyles,
      label,
      onPress,
      disabled
    } = this.props;

    const mergedButtonStyles = {
      ...button.roundButton,
      ...buttonStyles
    };
    const mergedTextStyles = {
      ...text.primaryText,
      ...textStyles
    };

    return (
      <TouchableOpacity
        style={mergedButtonStyles}
        onPress={onPress}
        disabled={disabled}
      >
        <Text style={mergedTextStyles}>{label}</Text>
      </TouchableOpacity>
    );
  }
}
