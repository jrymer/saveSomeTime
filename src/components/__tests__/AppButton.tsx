import React from 'react';
import { StyleSheet } from 'react-native';
import renderer from 'react-test-renderer';

import { button, theme } from '../../styles';
import { AppButton } from '../AppButton';

// Note: test renderer must be required after react-native.
const styles = StyleSheet.create({
  butonStyles: {
    ...button.roundButton,
    backgroundColor: theme.primary,
    width: 100,
    height: 40
  },
  textStyles: {
    color: theme.font
  }
});

const props = {
  buttonStyles: styles.butonStyles,
  textStyles: styles.textStyles,
  onPress: () => true,
  label: 'label'
};

it('renders correctly', () => {
  const tree = renderer.create(
    < AppButton
      buttonStyles={props.buttonStyles}
      textStyles={props.textStyles}
      onPress={props.onPress}
      label={props.label}
    />).toJSON();
  expect(tree).toMatchSnapshot();
});
