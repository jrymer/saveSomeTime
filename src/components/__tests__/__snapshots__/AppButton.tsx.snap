// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#4a919e",
      "borderRadius": 20,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginTop": 10,
      "opacity": 1,
      "width": 100,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#FFF",
      }
    }
  >
    label
  </Text>
</View>
`;
